datacube/virtual/transformations.py

```diff
@@ -356,6 +357,16 @@ def result(output_var, output_desc):
 def year(time):
     return time.astype('datetime64[Y]')
 
+def fiscal_year(time):
+    """"
+    This function will support group-by financial years
+    """
+    df = pd.Series(time.values)
+    years = df.apply(lambda x: np.datetime64(str(x.to_period('Q-JUN').qyear))).values
+    ds = xr.DataArray(years, name='time', attrs=time.attrs, coords=time.coords, dims=time.dims)
```

Comment:
Looks good to me too.

What happens if instead of the two lines defining and modifying `ds` you try to do it in one go?
```python
ds = xr.DataArray(years, name='time', attrs=time.attrs, coords=years, dims=time.dims)
```
I _think_ that should work!

Reply:
The coordinates should be a dictionary so something like below will work but I'm not 100% sure if the `coords` can contain other variables than time which the following code will exclude:

 ds = xr.DataArray(years, name='time', attrs=time.attrs, coords=({"time": years}), dims=time.dims)

Reply:
How about writing a test - that way we know for sure if it works the way we expect.  :)

Reply:
Most of the existing time grouping transforms aren't covered by tests either, by they are trivial compared to this one.